public/app/features/dashboard/dashgrid/PanelHeader/PanelHeader.tsx

```diff
@@ -45,7 +46,9 @@ export class PanelHeader extends Component<Props, State> {
     const isFullscreen = false;
     const isLoading = false;
     const panelHeaderClass = classNames({ 'panel-header': true, 'grid-drag-handle': !isFullscreen });
-    const { panel, dashboard, timeInfo } = this.props;
+    const { panel, dashboard, timeInfo, scopedVars } = this.props;
+    const title = templateSrv.replaceWithText(panel.title, scopedVars);
```

Comment:
I think you can skip dependency to `templateSrv` in `PanelHeader` and simply pass `title` prop to `PanelHeader`. The `title` prop should take precedence over `panel.title` when specified.